### PR TITLE
Model aria better

### PR DESCRIPTION
**Pull request by jessebeach**

### CODE REVIEW

File: ariaAbstractRoles.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import rangeRole from './abstract/rangeRole';
+import roletypeRole from './abstract/roletypeRole';
+import sectionRole from './abstract/sectionRole';
+import sectionheadRole from './abstract/sectionheadRole';
+import selectRole from './abstract/selectRole';
+import structureRole from './abstract/structureRole';
+import widgetRole from './abstract/widgetRole';
+import windowRole from './abstract/windowRole';
+
+const ariaLiteralRoles = new Map([
```
Comment (evcohen) [inline ariaAbstractRoles.js]: s/ariaLiteralRoles/ariaAbstractRoles

File: ariaDpubRoles.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+import docPartRole from './dpub/docPartRole';
+import docPrefaceRole from './dpub/docPrefaceRole';
+import docPrologueRole from './dpub/docPrologueRole';
+import docPullquoteRole from './dpub/docPullquoteRole';
+import docQnaRole from './dpub/docQnaRole';
+import docSubtitleRole from './dpub/docSubtitleRole';
+import docTipRole from './dpub/docTipRole';
+import docTocRole from './dpub/docTocRole';
+
+const ariaLiteralRoles = new Map([
```
Comment (evcohen) [inline ariaDpubRoles.js]: s/ariaLiteralRoles/ariaDpubRoles

File: flow/roles.js [hunk 3]
```diff
@@ -6,15 +6,14 @@ type RoleDefinition = {
   abstract: boolean,
   baseConcepts: Array<RoleRelation>,
   childrenPresentational: boolean,
-  interactive: boolean,
-  props: Array<string>,
+  props: {[key: string]: mixed},
   relatedConcepts: Array<RoleRelation>,
-  requiredProps: Array<string>,
+  requiredProps: {[key: string]: mixed},
```
Comment (evcohen) [inline flow/roles.js]: comments and docs clarifying these properties would be helpful!

Comment (jessebeach) [inline flow/roles.js]: Righteo, can do.

**Comment by jessebeach** [comment]
Added commentary and tighter Flow type definitions. The goal here is to get them defined down to individual properties and then move all the typing into https://github.com/flowtype/flow-typed

**Pull request merged**